python/dgl/contrib/graph_partition_book.py

```diff
@@ -0,0 +1,254 @@
+import os
+import sys
+import numpy as np
+
+from .. import backend as F
+from ..base import NID, EID
+from ..data.utils import load_graphs, load_tensors
+
+class GraphPartitionBook:
+    def __init__(self, part_metadata, ip_config_file):
+        """Partition information.
+
+        Parameters
+        ----------
+        part_metadata : json object
+            metadata of partitioned graph, which is created by load_partition() API.
+        ip_config_file : str
+            path of IP configuration file.
+        """
+        self._part_meta = part_metadata
+        self._meta_data = [] # list[dict[str, any]]
+        self._nid2partid = None
+        self._eid2partid = None
+        self._partid2nids = []
+        self._partid2eids = []
+        self._nidg2l = []
+        self._eidg2l = []
+        # Get number of partitions
+        assert 'num_parts' in self._part_meta, "cannot get the number of partitions."
+        self._num_partitions = self._part_meta['num_parts']
+        assert self._num_partitions > 0, 'num_partitions cannot be a negative number.'
+        # Get part_files
+        self._part_files = []
+        for part_id in range(self._num_partitions):
+            assert 'part-{}'.format(part_id) in self._part_meta, "part-{} does not exist".format(part_id)
+            part_files = self._part_meta['part-{}'.format(part_id)]
+            self._part_files.append(part_files)
+        # Get part_graphs
+        self._part_graphs = []
+        for part_id in range(self._num_partitions):
+            graph = load_graphs(self._part_files[part_id]['part_graph'])[0][0]
+            self._part_graphs.append(graph)
```

Comment:
it's unlikely we can load all graph partitions.

Reply:
For now, get_partition(), nid2localnid(), and eid2localeid() will use the partitioned graph because we have a partid argument as input.